### PR TITLE
Remove 37 from CUDA compute archs

### PR DESCRIPTION
**Pull request by snnn**
### Description
To reduce CUDA package's size a little bit.  37 is for Tesla K80.  Azure's NC-series uses it, but in most cases CUDA can dynamic generate device code .

### CODE REVIEW

**Comment by jywu-msft** [comment]
how about removing 50 as well?  We would keep 52 which is M60

**Comment by snnn** [comment]
If we only remove 37, the binary size can be reduced from 344MB to 310MB. It doesn't seem much.

**Review by mszhanyi — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml [hunk 1]
```diff
@@ -28,13 +28,13 @@ jobs:
...
       clean: true
       submodules: recursive
-    - template: templates/get-docker-image-steps.yml
+    - template: get-docker-image-steps.yml
       parameters:
         Dockerfile: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_cuda11_6_tensorrt8_5
         Context: tools/ci_build/github/linux/docker
         DockerBuildArgs: "--network=host --build-arg POLICY=manylinux2014 --build-arg PLATFORM=x86_64 --build-arg DEVTOOLSET_ROOTPATH=/opt/rh/devtoolset-11/root --build-arg PREPEND_PATH=/opt/rh/devtoolset-11/root/usr/bin: --build-arg LD_LIBRARY_PATH_ARG=/opt/rh/devtoolset-11/root/usr/lib64:/opt/rh/devtoolset-11/root/usr/lib:/opt/rh/devtoolset-11/root/usr/lib64/dyninst:/opt/rh/devtoolset-11/root/usr/lib/dyninst:/usr/local/lib64 --build-arg BUILD_UID=$( id -u )"
         Repository: onnxruntimecuda114xtrt82build
-    - template: templates/set-version-number-variables-step.yml
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml:37]: same comment as above

File: tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml [hunk 2]
```diff
@@ -44,11 +44,11 @@ jobs:
...
           /opt/python/cp37-cp37m/bin/python3 /onnxruntime_src/tools/ci_build/build.py --build_dir /build --config Release \
-          --skip_submodule_sync --parallel --build_shared_lib ${{ parameters.buildJavaOption }} --use_tensorrt --tensorrt_placeholder_builder --cuda_version=$(CUDA_VERSION) --cuda_home=/usr/local/cuda-$(CUDA_VERSION) --cudnn_home=/usr --tensorrt_home=/usr --cmake_extra_defines CMAKE_CUDA_HOST_COMPILER=/opt/rh/devtoolset-11/root/usr/bin/cc 'CMAKE_CUDA_ARCHITECTURES=37;50;52;60;61;70;75;80'
+          --skip_submodule_sync --parallel --build_shared_lib ${{ parameters.buildJavaOption }} --use_tensorrt --tensorrt_placeholder_builder --cuda_version=$(CUDA_VERSION) --cuda_home=/usr/local/cuda-$(CUDA_VERSION) --cudnn_home=/usr --tensorrt_home=/usr --cmake_extra_defines CMAKE_CUDA_HOST_COMPILER=/opt/rh/devtoolset-11/root/usr/bin/cc 'CMAKE_CUDA_ARCHITECTURES=52;60;61;70;75;80'
         workingDirectory: $(Build.SourcesDirectory)
 
     - ${{ if eq(parameters.buildJava, true) }}:
-        - template: templates/java-api-artifacts-package-and-publish-steps-posix.yml
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml:51]: same comment as above

File: tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml [hunk 3]
```diff
@@ -57,15 +57,15 @@ jobs:
               libraryName: 'libonnxruntime.so'
               nativeLibraryName: 'libonnxruntime4j_jni.so'
 
-    - template: templates/c-api-artifacts-package-and-publish-steps-posix.yml
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml:60]: same comment as above

File: tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml [hunk 4]
```diff
@@ -57,15 +57,15 @@ jobs:
...
+    - template: c-api-artifacts-package-and-publish-steps-posix.yml
       parameters:
           buildConfig: 'Release'
           artifactName: ${{ parameters.artifactName }} 
           artifactNameNoVersionString: ${{ parameters.artifactNameNoVersionString }}  
           libraryName: 'libonnxruntime.so.$(OnnxRuntimeVersion)'
 
 
-    - template: templates/component-governance-component-detection-steps.yml
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml:68]: same

File: tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml [hunk 5]
```diff
@@ -57,15 +57,15 @@ jobs:
...
+    - template: component-governance-component-detection-steps.yml
       parameters :
         condition : 'succeeded'
-    - template: templates/clean-agent-build-directory-step.yml
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml:71]: here too

**Comment by tianleiwu** [comment]
Is this PR for reducing space to add 90 (H100) later?

**Comment by snnn** [comment]
> Is this PR for reducing space to add 90 (H100) later?

It is one of the reasons. For sure at some point we should add 90 there. Right now the top constraint is our GPU nuget package cannot be over 250MB.  Right now it is 244.55 MB. 

Now our packaging pipelines are using CUDA 11.6 which doesn't seem to have support for H100.  CUDA 11.8 should be good. Correct me if I was wrong.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**